akka-projection-testkit/src/test/scala/akka/projection/testkit/ProjectionTestKitSpec.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.projection.testkit
+
+import akka.{ Done, NotUsed }
+import akka.actor.ClassicActorSystemProvider
+import akka.actor.testkit.typed.scaladsl.{ ScalaTestWithActorTestKit, TestProbe }
+import akka.projection.Projection
+import akka.stream.{ KillSwitch, KillSwitches }
+import akka.stream.scaladsl.Source
+import akka.stream.scaladsl.{ Keep, Sink }
+import akka.stream.testkit.TestSubscriber
+import akka.stream.testkit.scaladsl.TestSink
+import org.scalatest.wordspec.AnyWordSpecLike
+import akka.actor.typed.scaladsl.adapter._
+
+import scala.concurrent.{ Future, Promise }
+
+class ProjectionTestKitSpec extends ScalaTestWithActorTestKit with AnyWordSpecLike {
+
+  val projectionTestKit: ProjectionTestKit = ProjectionTestKit(testKit)
+
+  "ProjectionTestKit" must {
+
+    "run a projection" in {
+      val probe = testKit.createTestProbe[Int]("projection-probe")
+      val prj = TestProjection(Source(1 to 5), probe)
+
+      val stopped =
+        projectionTestKit.run(prj) {
+          probe.expectMessage(1)
+          probe.expectMessage(2)
+          probe.expectMessage(3)
+          probe.expectMessage(4)
+          probe.expectMessage(5)
```

Comment:
I'm not sure I understand this usage? The `testFunc` will be called multiple times if it throws since it's wrapped in `awaitAssert`.

Reply:
Yes. The `testFunc` (will rename to `assertFunc`) is where the user asserts that the events reached their projection target. 

For instance, I would check if my model was updated as expected with the events that were emitted. 

I may be emitting 10 events and want to observe that my model is updated after I processed the 10 events. 

In the Slick Projection PR and on previous exploration PRs, I was using `ScalaTest.eventually`. The goal here is to have the same functionality but using a TestProbe instead. 

The example here may be misleading because there is no view model and I'm using a probe to observe the events flowing through the projection.

Reply:
@patriknw, I added some extra test to make it clear and avoid using the test probe. 

We may still need to have something to prove that the projection is not running.